pkg/controller/scansettingbinding/scansettingbinding_controller.go

```diff
@@ -307,7 +312,7 @@ func (r *ReconcileScanSettingBinding) Reconcile(request reconcile.Request) (reco
 
 	if found.Status.Phase == compliancev1alpha1.PhaseDone {
 		reqLogger.Info("Generating events for scansettingbinding")
-		common.GenerateEventForResult(r.recorder, instance, instance, found.Status.Result)
+		common.GenerateEventForResult(r.Recorder, instance, instance, found.Status.Result)
```

Comment:
Throwing this here for now since I can't leave a comment closer to line 388, but there must still be some issues with the roles or permissions:

```{"level":"info","ts":1660164557.0453293,"msg":"Observed a panic in reconciler: runtime error: invalid memory address or nil pointer dereference","controller":"tailoredprofile","controllerGroup":"compliance.openshift.io","controllerKind":"TailoredProfile","scanSettingBinding":{"name":"fedramp-moderate","namespace":"openshift-compliance"},"namespace":"openshift-compliance","name":"fedramp-moderate","reconcileID":"c4fc7ab7-c4f7-4ae8-95be-350ba0dc9475"}
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x90 pc=0x522670]

goroutine 856 [running]:
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile.func1()
        /go/src/github.com/openshift/compliance-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:118 +0x1f4
panic({0x1df83c0, 0x36a7b80})
        /usr/local/go/src/runtime/panic.go:1038 +0x215
regexp.(*Regexp).doExecute(0xc002c55440, {0x0, 0x0}, {0x0, 0x208, 0x0}, {0xc002ccd876, 0x0}, 0x0, 0x0, ...)
        /usr/local/go/src/regexp/exec.go:527 +0x90
regexp.(*Regexp).doMatch(...)
        /usr/local/go/src/regexp/exec.go:514
regexp.(*Regexp).MatchString(...)
        /usr/local/go/src/regexp/regexp.go:525
github.com/ComplianceAsCode/compliance-operator/pkg/controller/scansettingbinding.(*ReconcileScanSettingBinding).validateRoles(0xc0006fe7c0, 0xc002d1ce70)
        /go/src/github.com/openshift/compliance-operator/pkg/controller/scansettingbinding/scansettingbinding_controller.go:388 +0x149
github.com/ComplianceAsCode/compliance-operator/pkg/controller/scansettingbinding.(*ReconcileScanSettingBinding).applyConstraint(0x1e952e0, 0xc002720780, 0xc000902000, 0x2448ab0, {{0x2448ab0, 0xc00064a880}, 0xc0009f66e0})
        /go/src/github.com/openshift/compliance-operator/pkg/controller/scansettingbinding/scansettingbinding_controller.go:354 +0x276
github.com/ComplianceAsCode/compliance-operator/pkg/controller/scansettingbinding.(*ReconcileScanSettingBinding).Reconcile(0xc0006fe7c0, {0xc000100000, 0xc000bed560}, {{{0xc00077f908, 0x18}, {0xc000940db0, 0x40ee87}}})
        /go/src/github.com/openshift/compliance-operator/pkg/controller/scansettingbinding/scansettingbinding_controller.go:237 +0x133e
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0x2429e08, {0x2429eb0, 0xc000bed560}, {{{0xc00077f908, 0x1fb4220}, {0xc000940db0, 0x4108fd}}})
        /go/src/github.com/openshift/compliance-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:121 +0xd1
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc000509360, {0x2429e08, 0xc0006fe440}, {0x1e91580, 0xc0009e60e0})
        /go/src/github.com/openshift/compliance-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:320 +0x33c
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc000509360, {0x2429e08, 0xc0006fe440})
        /go/src/github.com/openshift/compliance-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:273 +0x205
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()
        /go/src/github.com/openshift/compliance-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:234 +0x85
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
        /go/src/github.com/openshift/compliance-operator/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:230 +0x36f
```